vignettes/datatable-importing.Rmd

```diff
@@ -195,6 +195,88 @@ For more canonical documentation of defining packages dependency check the offic
 
 Some of internally used C routines are now exported on C level thus can be used in R packages directly from their C code. See [`?cdt`](https://rdatatable.gitlab.io/data.table/reference/cdt.html) for details and [Writing R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html) _Linking to native routines in other packages_ section for usage.
 
-## Importing from non-R Applications {non-r-API}
+## Importing from non-r Applications {non-r-api}
 
 Some tiny parts of `data.table` C code were isolated from the R C API and can now be used from non-R applications by linking to .so / .dll files. More concrete details about this will be provided later; for now you can study the C code that was isolated from the R C API in [src/fread.c](https://github.com/Rdatatable/data.table/blob/master/src/fread.c) and [src/fwrite.c](https://github.com/Rdatatable/data.table/blob/master/src/fwrite.c).
+
+## How to convert your Depends dependency on data.table to Imports
+
+To convert a `Depends` dependency on `data.table` to an `Imports` dependency in an R package, follow these steps:
+
+### 1. Update the DESCRIPTION File
+In your package's `DESCRIPTION` file, change the entry from `Depends` to `Imports`. This involves moving `data.table` from the `Depends` field to the `Imports` field.
+
+**Before:**
+```r
+Depends:
+    R (>= 3.5.0),
+    data.table
+Imports:
+```
+
+**After:**
+```r
+Depends:
+    R (>= 3.5.0)
+Imports:
+    data.table
+```
+
+### 2. Modify the NAMESPACE File
+Ensure that any functions from `data.table` that you use in your package are imported. This can be done either by explicitly importing specific functions or by importing the entire package but specific imports are considered best practice.
+
+#### Importing Specific Functions
+If you use specific functions like `data.table()`, `setDT()`, or `merge()`, you can import them individually in the `NAMESPACE` file:
+
+```r
+importFrom(data.table, data.table, setDT, merge)
+```
+
+#### Bulk Import
+Alternatively, you can import all functions from `data.table` at once, though this is generally not recommended as best practice:
+```r
+import(data.table)
+```
+
+By including these import statements in your `NAMESPACE` file, you ensure that the necessary functions are available for your package to use.
+
+### 3. Update Your R code files
+When you move a package from `Depends` to `Imports`, it will no longer be automatically attached when your package is loaded. This means you need to refer to `data.table` functions explicitly using the `data.table::` prefix, or import the functions in your scripts. However, if you import `data.table` in your Namespace, you won't need to make any changes in your R scripts.
```

Comment:
this is still incorrect.
did you copy this information from somewhere?
I don't think any changes should be necessary in R code files actually. (as long as data.table is imported)

Reply:
Not sure what you mean here Toby. I think Nitish is talking about an analysis script, not package code. E.g.

```r
library(MyPkgDependsDataTable)
data.table(a = 1, b = MyFunction(2))
```

That script runs perfectly fine under `Depends` mode, but will stop working when switching to `Imports`.

That said, I think the easiest solution is to simply attach data.table explicitly:

```r
library(data.table)
library(MyPkgDependsDataTable)
data.table(a = 1, b = MyFunction(2))
```

This is much easier when doing one package at a time. It can be a real bear trap when switching multiple packages out of your `Depends` at once, because there can be subtle assumptions about the attach order (resulting `search()` path) that we'd need to respect to be _100%_ back-compatible. That problem is very difficult, but especially because data.table doesn't itself have any dependencies, I think we can avoid it in this discussion.

Reply:
if we talk about an R analysis script (not R package code) then that needs to be clear, because the other discussion was about converting an R package

Reply:
NB: it's still relevant in this vignette because package vignettes will be affected. So will other R files outside R/ directory like tests and demos